### PR TITLE
Polynomial Regression Modeling

### PR DESCRIPTION
**Pull request by seyerly**
Implemented new functionality for polynomial regression modeling.

### CODE REVIEW

File: polyssifier/polyssifier.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def _scorer(clf, X, y):
             try:
                 ypred = ypred[:, 1]
             except:
-                print('predict proba return shape {}'.format(yprob.shape))
+                print('predict proba return shape')
```
Comment (alvarouc) [inline polyssifier/polyssifier.py]: why is this changed?

File: polyssifier/polyssifier.py [hunk 2]
```diff
@@ -289,8 +289,45 @@ def fit_clf(args, clf_name, val, n_fold, project_name, save, scoring):
             clf)  # fitted clf
 
 
+def create_multivariate(data, degree):
```
Comment (alvarouc) [inline polyssifier/polyssifier.py]: data is already multivariate, change to create_polynomial

File: polyssifier/polyssifier.py [hunk 3]
```diff
@@ -289,8 +289,45 @@ def fit_clf(args, clf_name, val, n_fold, project_name, save, scoring):
...
+    width_exponential_matrix = data.shape[1]
+
+    for i in range(degree):
+        to_add_in = exponent_matrix(data, (i + 1))
+        for j in range(height_exponential_matrix):
+            for k in range(width_exponential_matrix):
+                to_pass_through.itemset((j, k + i * width_exponential_matrix), (to_add_in.item(j, k)))
+    return to_pass_through
+
+def exponent_matrix(matrix, exponent):
```
Comment (alvarouc) [inline polyssifier/polyssifier.py]: This function is not needed, you can just make the matrix a numpy array and do
`matrix ** exponent`

File: tests/test_matrix_operations.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    matrix_2pow = np.matrix('1 4; 9 16')
+    matrix_3pow = np.matrix('1 8; 27 64')
+    matrix_4pow = np.matrix('1 16; 81 256')
+    assert (matrix == exponent_matrix(matrix, 1)).all()
+    assert (matrix_2pow == exponent_matrix(matrix, 2)).all()
+    assert (matrix_3pow == exponent_matrix(matrix, 3)).all()
+    assert (matrix_4pow == exponent_matrix(matrix, 4)).all()
+
+@pytest.mark.medium
+def test_create_multivariate():
```
Comment (alvarouc) [inline tests/test_matrix_operations.py]: test_create_polynomial

File: tests/test_matrix_operations.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    matrix = np.matrix('5 6; 7 8')
```
Comment (alvarouc) [inline tests/test_matrix_operations.py]: Please use numpy arrays instead of matrix. This way it will be consistent across all the code

**Review by alvarouc — CHANGES_REQUESTED** [review]
Thanks for working on this, please address the comments and re-submit.

File: polyssifier/polyssifier.py (hunk 1 above)
Comment (seyerly) [reply polyssifier/polyssifier.py]: Did you mean to have ypred.shape orignally? yprob isn't initialized or referenced anywhere inside of the scope of _scorer so if the exception was ever caught the print statement would cause the program to crash. I took it out because it would have caused a crash but now that I look at it it seems like a typo as ypred.shape is called many other times throughout the _scorer function.

Comment (alvarouc) [reply polyssifier/polyssifier.py]: oh, I see. Change it to `ypred.shape` instead of yprob

Comment (seyerly) [reply polyssifier/polyssifier.py]: Will do.

File: polyssifier/polyssifier.py (hunk 2 above)
Comment (seyerly) [reply polyssifier/polyssifier.py]: Will do.

File: polyssifier/polyssifier.py (hunk 3 above)
Comment (seyerly) [reply polyssifier/polyssifier.py]: Will do.

File: tests/test_matrix_operations.py (hunk 5 above)
Comment (seyerly) [reply tests/test_matrix_operations.py]: I will change to arrays.

File: polyssifier/polyssifier.py (hunk 2 above)
Comment (seyerly) [reply polyssifier/polyssifier.py]: Fixed.

File: polyssifier/polyssifier.py (hunk 3 above)
Comment (seyerly) [reply polyssifier/polyssifier.py]: Fixed.

File: tests/test_matrix_operations.py (hunk 4 above)
Comment (seyerly) [reply tests/test_matrix_operations.py]: Fixed.

File: tests/test_matrix_operations.py (hunk 5 above)
Comment (seyerly) [reply tests/test_matrix_operations.py]: Fixed.

**Review by alvarouc — APPROVED** [review]
Thanks for the good work!
